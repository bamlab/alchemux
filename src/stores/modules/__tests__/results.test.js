import {
  default as reducer,
  resultsSelector,
  STORE_KEY,
  fetchSuccesResultsAction,
} from '../results';

describe('entities', () => {
  describe('reducer', () => {
    it('should have an initial state', () => {
      expect(reducer(undefined, { type: '@INIT' })).toMatchSnapshot();
    });

    it('should store given ids', () => {
      const action = {
        type: 'FETCH_SUCCESS',
        payload: [1, 2, 3],
        meta: { entity: 'foo', queryType: 'bar' },
      };
      expect(reducer(undefined, action)).toMatchSnapshot();
    });

    it('should store uniq ids', () => {
      const action = {
        type: 'FETCH_SUCCESS',
        payload: 1,
        meta: { entity: 'foo', queryType: 'bar' },
      };
      expect(reducer(undefined, action)).toMatchSnapshot();
    });
  });

  describe('module', () => {
    it('should get the ids', () => {
      const results = [1, 2, 3];
      const action = fetchSuccesResultsAction(results, 'notes', 'list');
      const state = reducer(undefined, action);
      const globalState = { [STORE_KEY]: state };
      expect(resultsSelector(globalState, 'notes', 'list')).toBe(results);
      expect(resultsSelector(globalState, 'users', 'list')).toBeUndefined();
      expect(resultsSelector(globalState, 'notes', 'one')).toBeUndefined();
    });
    it('should get the id', () => {
      const action = fetchSuccesResultsAction(1, 'notes', 'single');
      const state = reducer(undefined, action);
      const globalState = { [STORE_KEY]: state };
      expect(resultsSelector(globalState, 'notes', 'single')).toBe(1);
    });
  });
});
