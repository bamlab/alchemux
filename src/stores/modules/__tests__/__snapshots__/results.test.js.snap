// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entities reducer should have an initial state 1`] = `Object {}`;

exports[`entities reducer should store given ids 1`] = `
Object {
  "foo-bar": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`entities reducer should store uniq ids 1`] = `
Object {
  "foo-bar": 1,
}
`;
